src/Symfony/Component/Form/Extension/Core/DataAccessor/PropertyPathAccessor.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Extension\Core\DataAccessor;
+
+use Symfony\Component\Form\DataAccessorInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\PropertyAccess\Exception\AccessException;
+use Symfony\Component\PropertyAccess\Exception\UninitializedPropertyException;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+
+/**
+ * Writes and reads values to/from an object or array using property path.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ * @author Bernhard Schussek <bschussek@gmail.com>
+ */
+class PropertyPathAccessor implements DataAccessorInterface
+{
+    private $propertyAccessor;
+    private $fallbackAccessor;
+
+    public function __construct(PropertyAccessorInterface $propertyAccessor = null, DataAccessorInterface $fallbackAccessor = null)
+    {
+        $this->propertyAccessor = $propertyAccessor ?: PropertyAccess::createPropertyAccessor();
+        $this->fallbackAccessor = $fallbackAccessor;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getValue($data, FormInterface $form)
+    {
+        if (null !== $propertyPath = $form->getPropertyPath()) {
+            return $this->getPropertyValue($data, $propertyPath);
+        }
+
+        return $this->fallbackAccessor->getValue($data, $form);
```

Comment:
(Does not apply if we remove fallback accessors completely).
Since it's possible to create an instance without a fallbackAccessor present, this line will produce a fatal error.

Reply:
NVM, moved to Chain pattern.